dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -463,13 +468,16 @@ public void run(DockstoreWebserviceConfiguration configuration, Environment envi
         // optional CORS support
         // Enable CORS headers
         // final FilterRegistration.Dynamic cors = environment.servlets().addFilter("CORS", CrossOriginFilter.class);
-        final FilterHolder filterHolder = environment.getApplicationContext().addFilter(CrossOriginFilter.class, "/*", EnumSet.of(REQUEST));
+        CORS_ENDPOINTS.stream().forEach(urlContext -> {
+            FilterHolder filterHolder = environment.getApplicationContext().addFilter(CrossOriginFilter.class, urlContext, EnumSet.of(REQUEST));
+
+            filterHolder.setInitParameter(ACCESS_CONTROL_ALLOW_METHODS_HEADER, "GET,POST,DELETE,PUT,OPTIONS,PATCH");
+            filterHolder.setInitParameter(ALLOWED_ORIGINS_PARAM, "*");
+            filterHolder.setInitParameter(ALLOWED_METHODS_PARAM, "GET,POST,DELETE,PUT,OPTIONS,PATCH");
+            filterHolder.setInitParameter(ALLOWED_HEADERS_PARAM,
+                    "Authorization, X-Auth-Username, X-Auth-Password, X-Requested-With,Content-Type,Accept,Origin,Access-Control-Request-Headers,cache-control");
```

Comment:
We don't use X-Auth-Username and X-Auth-Password, AFAIK. I don't know why they got put there.

Also, although TRS is mainly read-only anyway, maybe we shouldn't allow the Authorization header until/if there is a need for it.

Also, it might be worth taking a look at all the parameters typically sent by the browser that we might allow, e.g., `Accept-Encoding`, which allows for zipped responses; I think we'd want to let users do that (unless it's already allowed by default).

Reply:
These changes were accepted